File: java/374猜数字大小.java
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */

public class Solution extends GuessGame {
    public int guessNumber(int n) {
        int left = 1;
        int right = n;
        int mid = 0;
        while(left < right){
            mid = (left + right) >>> 1;
            if(guess(mid) == 1){left = mid + 1;}
            else if(guess(mid) == -1){right = mid;}
            else{return mid;}
        }
        return left;
    }
}

File: python/374猜数字大小.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num: int) -> int:

# 遍历法 ： 超出时间限制
# class Solution:
#     def guessNumber(self, n):
#         pick = 1
#         while guess(pick) != 0:
#             pick = pick + 1
#         return pick

# 二分查找法
class Solution:
    def guessNumber(self, n: int) -> int:
        left = 1
        right = n
        while left < right:
            # 中位数
            mid = (left + right) >> 1
            if guess(mid) == 1:
                left = mid + 1
            elif guess(mid) == -1:
                right = mid
            else:
                return mid
        return left



